src/inference/src/dev/core_impl.cpp

```diff
@@ -1076,8 +1076,20 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
         // explicitly enabled Auto-Batching
         auto pos = deviceName.find_first_of(":");
         if (pos == std::string::npos)
-            return model;  // BATCH device is already configured via the config
-        deviceNameWithBatchSize = deviceName.substr(pos + 1);
+            return model;
+        // BATCH device is already configured via the config
+        // handle the hw plugin doesn't support optimal_batch_size & user doesn't assign the batch size, such as
+        // "BATCH:CPU" for this situation will assign the batch size to 1. ("BATCH:CPU" -> "BATCH:CPU(1)")
+        if (deviceName.find("(") == std::string::npos) {
+            auto supported_properties =
+                ICore::get_property(deviceName.substr(pos + 1, deviceName.find("(")), ov::supported_properties, {});
+            if (std::find(supported_properties.begin(), supported_properties.end(), ov::optimal_batch_size) ==
```

Comment:
should we just exit from this function in this place? because auto batch is applied, but HW device does not support it and instead of adding dummy `(1)` at the end, we could just exit from the function